test/hack/soak/main.go

```diff
@@ -0,0 +1,59 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/samber/lo"
+)
+
+// Goal: list all eks cluster that will need run a set of Integration testing
+
+func main() {
+	ctx := context.Background()
+	cfg := lo.Must(config.LoadDefaultConfig(ctx))
+	eksClient := eks.NewFromConfig(cfg)
+
+	var outputList []string
+	createNewCluster := true
+
+	clusters := lo.Must(eksClient.ListClusters(ctx, &eks.ListClustersInput{}))
+
+	for _, cluster := range clusters.Clusters {
+		clusterDetails := lo.Must(eksClient.DescribeCluster(ctx, &eks.DescribeClusterInput{Name: aws.String(cluster)}))
+		if clusterDetails.Cluster.CreatedAt.YearDay() == time.Now().YearDay() {
+			createNewCluster = false
+		}
+
+		if strings.HasPrefix(cluster, "soak-periodic-") {
+			outputList = append(outputList, cluster)
+		}
+	}
+
+	if createNewCluster {
+		outputList = append(outputList, "createNewCluster")
+	}
+
+	output, _ := json.Marshal(outputList)
+	fmt.Println(string(output))
```

Comment:
Should just return the list of cluster names and it just doesn't do anything for an empty/non-empty list.

Reply:
We would need some way of indicating that we need to create a cluster. This would be the best way to go forward. To make the e2e logic cleaner, that will treat `createNewCluster` as an empty string